File: 2024_oct/1497 Check If Array Pairs Are Divisible by k.py
class Solution:
    def canArrange(self, arr: List[int], k: int) -> bool:
        my_dict = defaultdict(int)
        for elem in arr:
            rem = elem % k
            my_dict[rem] += 1
        
        for key, values in my_dict.items():
            if key != 0 and my_dict[key] != my_dict[k-key]:
                return False
            if key == 0 and my_dict[key] % 2 != 0:
                return False
        return True

File: 2024_oct/1497 Check If Array Pairs Are Divisible by k.java
class Solution {
    public boolean canArrange(int[] arr, int k) {
        
        int[] remainderFreq = new int[k];
        
        for (int num : arr) {
            int remainder = (num % k + k) % k; 
            remainderFreq[remainder]++;
        }
        
        for (int i = 0; i <= k / 2; i++) {
            if (i == 0) {
                if (remainderFreq[i] % 2 != 0) return false;
            } else {
                if (remainderFreq[i] != remainderFreq[k - i]) return false;
            }
        }
        
        return true;
    }
}